src/lib.rs

```diff
@@ -156,6 +173,16 @@ mod tests {
         ChaCha20Rng::from_entropy()
     }
 
+    #[test]
+    fn test_combine_signature_shares_inner() {
+        let res = combine_signature_shares_inner::<Bls12381G2Impl>(&vec![
+            r#"{"ProofOfPossession":"0292575ada70fcd604499062a0518c61726d3172f3d58bcafdaed1475b7beee4f4e46a88e326378961dd877d1ca0db2989170535dbf5c773fa9933fd8ca0260b9e762ee7a81ea2957e7cb3211f1e43973eab8e418fde6e316c35e9bb3221bd50af"}"#.to_string(),
+            r#"{"ProofOfPossession":"01b6ce4a8bc406706315287bc7615c225206bfc26e75b532ec7299b9ad538b184b7bf91ba953199f207c8502806276f14c0b8864343888b060f8d4c8f58514563d788b54344401470178621500d053cbfc5322baa15d1125c830762a8a5b07a273"}"#.to_string(),
+            r#"{"ProofOfPossession":"038c6a9a1aa415b49cffced6fa2054718bfda766a160e4968dc72cad861cfdbe2248eb0b93ebde19c026383692c0ad05311335fa9d96e5c58e1d004059f1cfd4429d3b2e3a2284b9060dfb9c97996feec169c37e20a32ffce93e4ee44af53c9587"}"#.to_string(),
+        ]);
+        assert!(res.is_ok());
```

Comment:
should we assert on the structure of the signature returned? is a boolean check enough?

Reply:
Think so, given that the same logic is actually tested in the other unit tests too in a more E2E style